pkg/userdata/centos/provider.go

```diff
@@ -121,7 +122,8 @@ func (p Provider) UserData(
 	if err != nil {
 		return "", fmt.Errorf("failed to execute user-data template: %v", err)
 	}
-	return b.String(), nil
+	clean := regexp.MustCompile(`(?m)^[ \t]+$`).ReplaceAllString(b.String(), "")
```

Comment:
1. Won't this also break data which requires tabs ?

If not, add a comment explaining why its necessary

Reply:
Doesn't remove leading tabs in front of content. Only reduces lines with only tabs and/or spaces (generated by a template plugin) to an empty line.